src/qibolab/execution_parameters.py

```diff
@@ -37,6 +37,16 @@ class AveragingMode(Enum):
     """SEQUENTIAL: Worse averaging for noise[Avoid]"""
 
 
+class BatchingMode(Enum):
+    """Data acquisition from hardware."""
+
+    SEQUENCES = auto()
+    """Demodulate, integrate the waveform and discriminate among states based
+    on the voltages."""
+    DURATION = auto()
+    """Demodulate and integrate the waveform."""
```

Comment:
As said, this depends more on the instrument than on what you're executing. Maybe you could already merge the two batching modes into the single loop, and avoid extra parameters.

Reply:
The idea is that this should be device independent and more related to: Is the sequence constant during the experiment or does it stay the same ?

Reply:
I guess devices have different memory limitations, so the batching limits are really device-dependent.

It is true that which one matter the most for sure depend on the specific sequence. For this reason, if you put them in AND you're just limited by the most constraining one for the given sequence.